audit.go

```diff
@@ -0,0 +1,162 @@
+	ExecutionContext ExecutionContext `json:"execution_context,omitempty"`
+	Actors           []Actor          `json:"actors,omitempty"`
+	Method           Method           `json:"method,omitempty"`
+	RootResource     Resource         `json:"root_resource,omitempty"`
+	Action           string           `json:"action,omitempty"`
+	Details          Details          `json:"details,omitempty"`
+}
+
+// ResponseMetadata contains information about the response.
+type ResponseMetaData struct {
+	Messages []string `json:"messages,omitempty"`
+}
+
+// ExecutionContext contains information about the action execution context.
+type ExecutionContext struct {
+	RequestID     string `json:"request_id,omitempty"`
+	RemoteAddress string `json:"remote_address,omitempty"`
+}
+
+// Actor is the actor who carried out the action.
+type Actor APIObject
+
+// Resource contains information on the resource that was audited.
+type Resource APIObject
+
+// Method contains information on the method used to perform the action.
+type Method struct {
+	Description    string `json:"description,omitempty"`
+	TruncatedToken string `json:"truncated_token,omitempty"`
+	Type           string `json:"type,omitempty"`
+}
+
+// Details contain additional information about the action or the resource
+// that has been audited.
+type Details struct {
+	Resource Resource `json:"resource,omitempty"`
+	// A set of fields that have been affected.
+	// The fields that have not been affected MAY be returned.
+	Fields []Field `json:"fields,omitempty"`
+	// A set of references that have been affected.
+	References []Reference `json:"references,omitempty"`
+}
+
+// Field contains information about the resource field that have been affected.
+type Field struct {
+	Name        string `json:"name,omitempty"`
+	Description string `json:"description,omitempty"`
+	Value       string `json:"value,omitempty"`
+	BeforeValue string `json:"before_value,omitempty"`
+}
+
+// Reference contains information about the reference that have been affected.
+type Reference struct {
+	Name        string      `json:"name,omitempty"`
+	Description string      `json:"description,omitempty"`
+	Added       []APIObject `json:"added,omitempty"`
+	Removed     []APIObject `json:"removed,omitempty"`
+}
+
+// ListAuditRecords lists audit trial records matching provided query params
+// or default criteria.
+func (c *Client) ListAuditRecords(ctx context.Context, o ListAuditRecordsOptions) (*ListAuditRecordsResponse, error) {
+	v, err := query.Values(o)
+	if err != nil {
+		return nil, err
+	}
+
+	u := fmt.Sprintf("%s?%s", auditBaseURL, v.Encode())
+	resp, err := c.get(ctx, u)
+	if err != nil {
+		return nil, err
+	}
+
+	var result ListAuditRecordsResponse
+	if err = c.decodeJSON(resp, &result); err != nil {
+		return nil, err
+	}
+
+	return &result, nil
+
+}
+
+// ListAuditRecordsPaginated lists audit trial records matching provided query
+// params or default criteria, processing paginated responses.
```

Comment:
Could you add these to sentences to the end of this comment:

```
The include function decides whether or not to include a specific AuditRecord in the final result. If the include function is nil, all audit records from the API are included by default.
```

The rest looks good!